gibbon-compiler/src/Gibbon/Passes/InferLocations.hs

```diff
@@ -576,28 +579,30 @@ inferExp env@FullEnv{dataDefs}
 
     LitSymE s -> return (lc$ LitSymE s, SymTy, [])
 
-    AppE f ls arg ->
+    AppE f _ args ->
         do let arrty = lookupFEnv f env
-           valTy <- freshTyLocs $ arrOut arrty
-           argTy <- freshTyLocs $ arrIn arrty
-           argDest <- destFromType' argTy
-           (arg',aty,acs) <- inferExp env arg argDest
+           valTy    <- freshTyLocs $ arrOut arrty
+           -- /cc @vollmerm
```

Comment:
@vollmerm  the changes I made to `InferLocations` didn't break anything (I think). One thing we can do now is go over the things that deal with tuples and see if the new representation helps us fix any bugs. Ditto for Cursorize.

Reply:
Okay. I don't know of any tuple bugs at the moment in inferlocs but I can believe there probably are some.